File: iosInblo/iosInblo/App/calendar/EventType.swift
//
//  EventType.swift
//  iosInblo
//
//  Created by Jhondee Diaz on 2/2/22.
//  Copyright © 2022 orgName. All rights reserved.
//

import Foundation
import SwiftUI


enum EventOptions: CaseIterable {
    
    struct EventInfo {
        var name: String
        var colorDark: Color
        var colorDarker: Color
        var colorLight: Color
    }
    
    case final_cutoff
    case interm_cutoff
    case race_sched
    case farrier
    case stables_retire
    case stables_return
    case stable_related
    case business_trip
    case others
    
    public static func getEventInfoByEventType(eventType: String) -> EventOptions? {
        return EventOptions.allCases.first(where: { $0.info.name == eventType })
    }
    
    var info: EventInfo {
        switch self {
        case .final_cutoff:
            return EventInfo(name: "最終追切", colorDark: Color(hex: "#FFD033")!, colorDarker: Color(hex: "#F2C100")!, colorLight: Color(hex: "#FFF4C3")!)
            
        case .interm_cutoff:
            return EventInfo(name: "中間追切", colorDark: Color(hex: "#FFD033")!, colorDarker: Color(hex: "#F2C100")!, colorLight: Color(hex: "#FFF4C3")!)
        case .race_sched:
            return EventInfo(name: "レース予定", colorDark: Color(hex: "#EA7266")!, colorDarker: Color(hex: "#E55863")!, colorLight: Color(hex: "#FFDFDF")!)
        case .farrier:
            return EventInfo(name: "装蹄", colorDark: Color(hex: "#EA7266")!, colorDarker: Color(hex: "#E55863")!, colorLight: Color(hex: "#FFDFDF")!)
        case .stables_retire:
            return EventInfo(name: "退厩", colorDark: Color(hex: "#FFD033")!, colorDarker: Color(hex: "#F2C100")!, colorLight: Color(hex: "#FFF4C3")!)
        case .stables_return:
            return EventInfo(name: "帰厩", colorDark: Color(hex: "#22D143")!, colorDarker: Color(hex: "#44AD60")!, colorLight: Color(hex: "#DCF8E8")!)
        case .stable_related:
            return EventInfo(name: "厩舎関連", colorDark: Color(hex: "#22D143")!, colorDarker: Color(hex: "#44AD60")!, colorLight: Color(hex: "#DCF8E8")!)
        case .business_trip:
            return EventInfo(name: "出張・不在予定", colorDark: Color(hex: "#7D79E8")!, colorDarker: Color(hex: "#736FF1")!, colorLight: Color(hex: "#DCDEF8")!)
        case .others:
            return EventInfo(name: "その他", colorDark: Color(hex: "#7D79E8")!, colorDarker: Color(hex: "#736FF1")!, colorLight: Color(hex: "#DCDEF8")!)
        }
    }
}

File: androidInblo/src/main/java/com/colinjp/inblo/android/domain/model/EventType.kt
package com.colinjp.inblo.android.domain.model

import com.colinjp.inblo.android.R

enum class EventType(val type: String, val colorDark: Int, val colorDarker: Int, val colorLight: Int) {
    FINAL_CUT_OFF("最終追切", R.color.colorEventTypeDark_1, R.color.colorEventTypeDarker_1, R.color.colorEventTypeLight_1),
    INTERM_CUT_OFF("中間追切", R.color.colorEventTypeDark_1, R.color.colorEventTypeDarker_1, R.color.colorEventTypeLight_1),
    RACE_SCHED("レース予定", R.color.colorEventTypeDark_2, R.color.colorEventTypeDarker_2, R.color.colorEventTypeLight_2),
    FARRIER("装蹄", R.color.colorEventTypeDark_2, R.color.colorEventTypeDarker_2, R.color.colorEventTypeLight_2),
    STABLES_RETIRE("退厩", R.color.colorEventTypeDark_3, R.color.colorEventTypeDarker_3, R.color.colorEventTypeLight_3),
    STABLES_RETURN("帰厩", R.color.colorEventTypeDark_3, R.color.colorEventTypeDarker_3, R.color.colorEventTypeLight_3),
    STABLE_RELATED("厩舎関連", R.color.colorEventTypeDark_3, R.color.colorEventTypeDarker_3, R.color.colorEventTypeLight_3),
    BUSINESS_TRIP("出張・不在予定", R.color.colorEventTypeDark_4, R.color.colorEventTypeDarker_4, R.color.colorEventTypeLight_4),
    OTHERS("その他", R.color.colorEventTypeDark_4, R.color.colorEventTypeDarker_4, R.color.colorEventTypeLight_4)
}